plugins/BEdita/Core/src/Model/Action/UpdateAssociatedAction.php

```diff
@@ -95,9 +95,10 @@ protected function prepareRelatedEntities($relatedEntities, EntityInterface $sou
         $prefix = sprintf('%s.', $junction->getAlias());
         $extraFields = [];
         foreach ($conditions as $field => $value) {
-            if (substr($field, 0, strlen($prefix)) === $prefix) {
-                $field = substr($field, strlen($prefix));
+            if (substr($field, 0, strlen($prefix)) !== $prefix) {
+                continue;
```

Comment:
Why  `$extraFields` is populated differently from before.

Reply:
It is populated in a different way because before we used to have only conditions on the junction table: `$conditions === ['ObjectRelations.relation_id' => 42]`).
Now we also have conditions on the target table: `$conditions === ['ObjectRelations.relation_id' => 42, 'Objects.deleted' => false]`.

While `relation_id = 42` should be added as an "extra field" to the junction entity, `deleted = false` should not.